lib/src/react_client/private_utils.dart

```diff
@@ -0,0 +1,231 @@
+  bool shouldWarn = false;
+
+  if (finalList.contains('shouldComponentUpdate')) {
+    finalList.remove('shouldComponentUpdate');
+    shouldWarn = true;
+  }
+
+  if (finalList.contains('componentDidUpdate')) {
+    finalList.remove('componentDidUpdate');
+    shouldWarn = true;
+  }
+
+  if (finalList.contains('render')) {
+    finalList.remove('render');
+    shouldWarn = true;
+  }
+
+  if (shouldWarn) {
+    window.console.warn("WARNING: Crucial lifecycle methods passed into "
+        "skipMethods. shouldComponentUpdate, componentDidUpdate, and render "
+        "cannot be skipped and will still be added to the new component. Please "
+        "remove them from skipMethods.");
+  }
+
+  return finalList;
+}
+
+/// Converts a list of variadic children arguments to children that should be passed to ReactJS.
+///
+/// Returns:
+///
+/// - `null` if there are no args and [shouldAlwaysBeList] is false
+/// - `[]` if there are no args and [shouldAlwaysBeList] is true
+/// - the single child if only one was specified
+/// - otherwise, the same list of args, will all top-level children validated
+dynamic generateChildren(List childrenArgs, {bool shouldAlwaysBeList = false}) {
+  var children;
+
+  if (childrenArgs.isEmpty) {
+    if (!shouldAlwaysBeList) return null;
+    children = childrenArgs;
+  } else if (childrenArgs.length == 1) {
+    if (shouldAlwaysBeList) {
+      final singleChild = react_client_utils.listifyChildren(childrenArgs.single);
+      if (singleChild is List) {
+        children = singleChild;
+      }
+    } else {
+      children = childrenArgs.single;
+    }
+  }
+
+  if (children is Iterable && children is! List) {
+    children = children.toList(growable: false);
+  }
+
+  if (children == null) {
+    children = shouldAlwaysBeList ? childrenArgs.map(react_client_utils.listifyChildren).toList() : childrenArgs;
+    markChildrenValidated(children);
+  }
+
+  return children;
+}
+
+/// Converts [props] into a [JsMap] that can be utilized with `React.createElement()`.
+JsMap generateJsProps(Map props,
+    {bool shouldConvertEventHandlers = true,
+    bool convertRefValue = true,
+    bool convertCallbackRefValue = true,
+    bool wrapWithJsify = true}) {
+  final propsForJs = JsBackedMap.from(props);
+
+  if (shouldConvertEventHandlers) convertEventHandlers(propsForJs);
+  if (convertRefValue) convertRefValue2(propsForJs, convertCallbackRefValue: convertCallbackRefValue);
+
+  return wrapWithJsify ? jsifyAndAllowInterop(propsForJs) : propsForJs.jsObject;
+}
+
+@Deprecated('6.0.0')
+InteropContextValue jsifyContext(Map<String, dynamic> context) {
+  var interopContext = new InteropContextValue();
+  context.forEach((key, value) {
+    // ignore: argument_type_not_assignable
+    setProperty(interopContext, key, new ReactDartContextInternal(value));
+  });
+
+  return interopContext;
+}
+
+/// A wrapper around [validateJsApi] that will return the result of a callback if React is present.
```

Comment:
#nit this comment makes it vague what happens if React is not present.
```suggestion
/// Validates that React is available (see [validateJsApi]) before returning the result of [computeReturn]).
```